zeroex/order_validator_test.go

```diff
@@ -291,23 +265,52 @@ func TestCalculateRemainingFillableTakerAmount(t *testing.T) {
 	assert.Equal(t, new(big.Int).Div(takerAssetAmount, big.NewInt(2)), remainingFillableTakerAssetAmount)
 }
 
-func orderWithCustomMakerAssetAmount(order Order, makerAssetAmount *big.Int) Order {
-	order.MakerAssetAmount = makerAssetAmount
-	return order
+func signedOrderWithCustomMakerAssetAmount(t *testing.T, signedOrder SignedOrder, makerAssetAmount *big.Int) SignedOrder {
+	signedOrderCopy := copySignedOrder(signedOrder)
```

Comment:
If we change `SignedOrder.Order` so that it's not a pointer, it would also simplify this code a lot. The `Order` would also be copied so you could remove the `copyOrder` and `copySignedOrder` functions.